services/src/main/java/org/keycloak/services/resources/admin/AdminRoot.java

```diff
@@ -94,19 +94,31 @@ public static UriBuilder adminBaseUrl(UriBuilder base) {
     @GET
     @Operation(hidden = true)
     public Response masterRealmAdminConsoleRedirect() {
-        String requestUrl = session.getContext().getUri().getRequestUri().toString();
         KeycloakUriInfo adminUriInfo = session.getContext().getUri(UrlType.ADMIN);
+        if (shouldRedirect(adminUriInfo)) {
+            RealmModel master = new RealmManager(session).getKeycloakAdminstrationRealm();
+            return Response.status(302).location(
+                    adminUriInfo.getBaseUriBuilder().path(AdminRoot.class).path(AdminRoot.class, "getAdminConsole").path("/").build(master.getName())
+            ).build();
+        }
+        return Response.status(Response.Status.NOT_FOUND).build();
+    }
+
+    boolean shouldRedirect(KeycloakUriInfo adminUriInfo) {
+        if (!isAdminConsoleEnabled()) {
+            return false;
+        }
+        KeycloakUriInfo frontEndUriInfo = session.getContext().getUri();
+        String frontEndUrl = frontEndUriInfo.getBaseUri().toString();
         String adminUrl = adminUriInfo.getBaseUri().toString();
-        String localAdminUrl = session.getContext().getUri(UrlType.LOCAL_ADMIN).getBaseUri().toString();
 
-        if (!isAdminConsoleEnabled() || (!requestUrl.startsWith(adminUrl) && !requestUrl.startsWith(localAdminUrl))) {
-            return Response.status(Response.Status.NOT_FOUND).build();
+        if (adminUrl.equals(frontEndUrl)) {
```

Comment:
This makes sense to me - the issue I reported would enter here, my frontend and admin are the same.

Reply:
I built a new `keycloak-999.0.0-SNAPSHOT.zip` and tested my edge case.

```
$ ./bin/kc.sh start-dev \
--hostname=https://banana.russdev
```

In another window I curled the `/admin/` endpoint and saw the results I was expecting

```
$ curl -v http://banana.russdev:8080/admin/
< HTTP/1.1 302 Found
< Location: https://banana.russdev/admin/master/console/
```

LGTM!